drape_frontend/apply_feature_functors.cpp

```diff
@@ -464,9 +464,8 @@ void ApplyPointFeature::ProcessPointRule(Stylist::TRuleWrapper const & rule)
     m_symbolRule = symRule;
   }
 
-  bool const isNode = (rule.m_rule->GetType() & drule::node) != 0;
   CaptionDefProto const * capRule = rule.m_rule->GetCaption(0);
-  if (capRule && isNode)
+  if (capRule)
```

Comment:
Hm, why did you decide that its obsolete?
This function is called from ApplyAreaFeature::ProcessAreaRule. And apply_for_type attr says: do not apply this caption to area features.

Reply:
UPD: Ok, you find some old comment about obsolete. Honestly, I'm not sure that comment is right. Can we investigate any rule with apply_for_type for the appropriate Feature?

Reply:
Sorry, I didn't mention it initially, but of course I examined the code and tested it on a few types (line/area/node) to ensure that `apply_for_type` does nothing indeed.

Code-wise, `m_type` could be either `node | way` (init value) or `node` (set by `apply_for_type: 1` in drules). There is no code that could assign it to just `way`.
Hence `isNode = (rule.m_rule->GetType() & drule::node) != 0` will be `true` ALWAYS.

And it doesn't make sense to check it for caption rules anyway, because kothic adds `apply_for_type: 1` to icon rules only.